irohad/ametsuchi/impl/flat_file/flat_file.cpp

```diff
@@ -139,6 +139,7 @@ std::string FlatFile::directory() const {
 }
 
 Identifier FlatFile::last_id() const {
+  reload();
   return (available_blocks_.empty()) ? 0 : *available_blocks_.rbegin();
```

Comment:
I expect this method to be constant and very light. To make multiple FS calls is not a good decision. This mode must be disconnectable. The best way I see is an option for FlatFile to skip write blocks. And independent mechanics, that scans the directory for new and not locked expected files, that can be read.